src/overlays/actors/ovl_Obj_Mure/z_obj_mure.c

```diff
@@ -26,43 +40,375 @@ const ActorInit Obj_Mure_InitVars = {
+
+s32 sMaxChildSpawns[] = { 12, 9, 8, 0 };
+
+s16 sSpawnActorIds[] = { ACTOR_EN_KUSA, 0, ACTOR_EN_FISH, ACTOR_EN_INSECT, ACTOR_EN_BUTTE };
+
+s16 sSpawnParams[] = { 0, 2, -1, 0, -1 };
+
+// sInitChain
+InitChainEntry D_80B99A8C[] = {
+    ICHAIN_F32(uncullZoneForward, 1200, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 200, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1200, ICHAIN_STOP),
+};
+
+ObjMureActionFunc sTypeFunc[] = { NULL, NULL, func_80B995A4, func_80B995A4, func_80B997CC };
+
+s32 func_80B98AA0(Actor* thisx, GlobalContext* globalCtx) {
+    ObjMure* this = THIS;
+    s32 result;
+
+    switch (this->type) {
+        case OBJMURE_TYPE_FISH:
+        case OBJMURE_TYPE_BUGS:
+        case OBJMURE_TYPE_BUTTERFLY:
+            Actor_ProcessInitChain(&this->actor, D_80B99A8C);
+            result = true;
+            break;
+        default:
+            osSyncPrintf("Error : カリングの設定がされていません。(%s %d)(arg_data 0x%04x)\n", "../z_obj_mure.c", 204,
+                         this->actor.params);
+            return false;
+    }
+    return result;
+}
+
+s32 func_80B98B1C(Actor* thisx, GlobalContext* globalCtx) {
+    if (!func_80B98AA0(thisx, globalCtx)) {
+        return false;
+    }
+    return true;
+}
+
+void ObjMure_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjMure* this = THIS;
+
+    this->chNum = (thisx->params >> 0xC) & 0x0F;
+    this->ptn = (thisx->params >> 8) & 0x07;
+    this->svNum = (thisx->params >> 5) & 0x03;
+    this->type = thisx->params & 0x1F;
+
+    if (this->ptn >= 4) {
+        osSyncPrintf("Error 群れな敵 (%s %d)(arg_data 0x%04x)\n", "../z_obj_mure.c", 237, thisx->params);
+        Actor_Kill(&this->actor);
+        return;
+    } else if (this->type >= 5) {
+        osSyncPrintf("Error 群れな敵 (%s %d)(arg_data 0x%04x)\n", "../z_obj_mure.c", 245, thisx->params);
+        Actor_Kill(&this->actor);
+        return;
+    } else if (!func_80B98B1C(thisx, globalCtx)) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+    this->actionFunc = ObjMure_InitialAction;
+    osSyncPrintf("群れな敵 (arg_data 0x%04x)(chNum(%d) ptn(%d) svNum(%d) type(%d))\n", thisx->params, this->chNum,
+                 this->ptn, this->svNum, this->type);
+    if (ObjMure_GetMaxChildSpawns(this) <= 0) {
+        osSyncPrintf("Warning : 個体数が設定されていません(%s %d)(arg_data 0x%04x)\n", "../z_obj_mure.c", 268,
+                     thisx->params);
+    }
+}
+
+void ObjMure_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+// gets the number of spawned actors
+s32 ObjMure_GetMaxChildSpawns(ObjMure* this) {
+    if (this->chNum == 0) {
+        return sMaxChildSpawns[this->ptn];
+    }
+    return this->chNum;
+}
+
+void ObjMure_GetSpawnPos(Vec3f* outPos, Vec3f* inPos, s32 ptn, s32 idx) {
+    if (ptn >= 4) {
+        osSyncPrintf("おかしなの (%s %d)\n", "../z_obj_mure.c", 307);
+    }
+    *outPos = *inPos;
+}
+
+// ObjMure_SpawnActors
+void ObjMure_SpawnActors0(ObjMure* this, GlobalContext* globalCtx) {
```

Comment:
```suggestion
void ObjMure_SpawnActors0(ObjMure* this, GlobalContext* globalCtx) {
```
I don't like the 0 but it's fine until further documentation, I guess